src/modules/user/middleware/newPassword.js

```diff
@@ -0,0 +1,48 @@
+import { databasePrisma } from "../../../prismaClient";
+import bcrypt from 'bcryptjs';
+import crypto from 'crypto';
```

Comment:
I chose crypto for generating reset tokens because it's lightweight and native to Node.js, making it ideal for this purpose. bcrypt is great for password hashing, but for token generation, crypto seemed more efficient. I also felt it was easier to keep track of what was happening when I split it up like this. But I can understand that it might seem or be inconsistent and don't mind changing it, any more thoughts on this? Will change it tomorrow if I don't hear anything :)